### PR TITLE
PolymerProxyMixin -> PolymerBase and update for new polymer

### PR DESCRIPTION
**Pull request by jakemac53**
closes https://github.com/dart-lang/polymer_interop/issues/6

### CODE REVIEW

File: lib/src/polymer_base.dart [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+        DocumentFragment,
+        TemplateElement;
+import 'dart:js' as js;
+import 'package:web_components/web_components.dart'
+    show CustomElementProxyMixin;
+
+/// A mixin to make it easier to interoperate with Polymer JS elements.
+///
+/// You must also mixin an implementation of the [CustomElementProxyMixin] class
+/// `web_components`. The implements statement here enforces that.
```
Comment (sigmundch) [inline lib/src/polymer_base.dart]: nit: enforces that => hints that?

(technically it doesn't enforce it - users can satisfy the 'implements' by adding an implementation of every member declared in CustomElementProxyMixin.)

File: lib/src/polymer_base.dart [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+///
+/// This is created from http://polymer.github.io/polymer/, but does not contain
+/// the following methods:
```
Comment (sigmundch) [inline lib/src/polymer_base.dart]: you might want to say why it doesn't

**Comment by sigmundch** [comment]
lgtm

File: lib/src/polymer_base.dart (hunk 1 above)
Comment (jakemac53) [inline lib/src/polymer_base.dart]: Reworded

File: lib/src/polymer_base.dart (hunk 2 above)
Comment (jakemac53) [inline lib/src/polymer_base.dart]: done, also added a link to issue page in case people do need one of these

**Pull request merged**